src/components/Articles.js

```diff
@@ -1,33 +1,37 @@
 import { useEffect } from "react";
 import { getArticles } from "../utils/api";
+import { Link } from "react-router-dom";
 
-function Articles ({articles, setArticles}) {
 
-    useEffect(() => {
-      getArticles().then((retreivedArticles) => {
-        setArticles(retreivedArticles);
-      });
-    }, []);
-
-  
-    return (
-      <main>
-        <ul className="Articles_container">
-          {articles.map((article) => {
-            return (
-                <li className="Articles_container_article" key={article.article_id}>
-                  <h3>Title: {article.title}</h3>
-                  <p>Author: {article.author}</p>
-                  <p>Created: {new Date(article.created_at).toGMTString()}</p>
-                  <p>Votes: {article.votes}</p>
-                  <p>Comment count: {article.comment_count}</p>
-                </li>
-            );
-          })}
-        </ul>
-      </main>
-    );
+function Articles({ articles, setArticles }) {
+  useEffect(() => {
+    getArticles().then((retreivedArticles) => {
+      setArticles(retreivedArticles);
+    });
+  }, [setArticles]);
```

Comment:
When do we want this useEffect to run?

setArticles is currently in the dependency array, so it will run whenever setArticles changes (which only happens when the first loads, as this is when useState is used to create it).  If we want it to run when the component loads, we can keep the dependency array empty)

Reply:
thanks! Have now fixed the dependency array in the Articles component useEffect, and have added loading state to Articles and SingleArticle components.